src/main/java/com/ldtteam/blockui/controls/EntityIcon.java

```diff
@@ -0,0 +1,157 @@
+package com.ldtteam.blockui.controls;
+
+import com.ldtteam.blockui.Pane;
+import com.ldtteam.blockui.PaneBuilders;
+import com.ldtteam.blockui.PaneParams;
+import com.ldtteam.blockui.Render;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.Tesselator;
+import com.mojang.math.Quaternion;
+import com.mojang.math.Vector3f;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.entity.EntityRenderDispatcher;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Control to render an entity as an icon
+ */
+public class EntityIcon extends Pane
+{
+    @Nullable
+    private Entity entity;
+    private int count = 1;
+    private float yaw = 30;
+    private float pitch = -10;
+    private float headyaw = 0;
+
+    public EntityIcon()
+    {
+        super();
+    }
+
+    public EntityIcon(@NotNull PaneParams params)
+    {
+        super(params);
+
+        final String entityName = params.getString("entity");
+        if (entityName != null)
+        {
+            setEntity(new ResourceLocation(entityName));
+        }
+
+        this.count = params.getInteger("count", this.count);
+        this.yaw = params.getFloat("yaw", this.yaw);
+        this.pitch = params.getFloat("pitch", this.pitch);
+        this.headyaw = params.getFloat("head", this.headyaw);
+    }
+
+    public void setEntity(@NotNull ResourceLocation entityId)
+    {
+        final EntityType<?> entityType = ForgeRegistries.ENTITIES.getValue(entityId);
+        if (entityType != null)
+        {
+            setEntity(entityType);
+        }
+        else
+        {
+            resetEntity();
+        }
+    }
+
+    public void setEntity(@NotNull EntityType<?> type)
+    {
+        final Entity entity = type.create(mc.level);
+
+        if (entity != null)
+        {
+            setEntity(entity);
+        }
+        else
+        {
+            resetEntity();
+        }
+    }
+
+    public void setEntity(@NotNull Entity entity)
+    {
+        this.entity = entity;
+        this.setHoverPane(null);

```

Comment:
This is no go, if hover instance of tooltip then change text (similar to itemicon logic), same in resetEntity

Reply:
k